handshake/crypto_setup_client_test.go

```diff
@@ -640,13 +640,15 @@ var _ = Describe("Client Crypto Setup", func() {
 		})
 
 		It("tries to escalate the crypto after receiving a diversification nonce", func(done Done) {
-			go cs.HandleCryptoStream()
-			time.Sleep(50 * time.Millisecond) // wait for the first maybeUpgradeCrypto to finish
+			go func() {
+				defer GinkgoRecover()
+				cs.HandleCryptoStream()
+				Fail("HandleCryptoStream should not have returned")
```

Comment:
Then we should fix all of those places at some point… And the first step to that is not allowing this for new tests ;)

Having goroutines running after a test finishes causes all kinds of issues, just have a look at the hoops I had to jump through for the tests in the main package. I really want to avoid that.